dnf-docker-test/features/steps/test_behave.py

```diff
@@ -103,6 +123,29 @@ def piecewise_compare(a, b):
     return sorted(a) == sorted(b)
 
 
+def sack_fill():
+    """
+    @return: sack with installed packages
+
+    """
+    with dnf.Base() as base:
+        base.fill_sack()
+        return base.sack
```

Comment:
the with statement is destroyed when the function is left. rather drop this factory method, please.
```python

class a(object):
    def __enter__(self):
        print 'enter'
        return 'obj'
    def __exit__(self ,type, value, traceback):
        print 'exit'
        return False

def f():
    with a() as s:
        return s

a = f()
print s
```